plotting_refactor/DataViewer.py

```diff
@@ -0,0 +1,172 @@
+from PlotModifiers import ModifierLinecolor, ModifierLinestyle, ModifierColormap, PlotModifier
+
+
+class DataViewer(QtWidgets.QWidget, Ui_DataViewer):
+    def __init__(self, main_window):
+        super(DataViewer, self).__init__()
+        self.setupUi(self)
+
+        self.main_window = main_window
+        self.datacollector = DataCollector()
+
+        self.dataTreeWidget = DataTreeWidget(self, self.datacollector)
+        self.plotTreeWidget = PlotTreeWidget(self)
+
+        self.cmdClose.clicked.connect(self.onShowDataViewer)
+        self.cmdAddModifier.clicked.connect(self.onAddModifier)
+        self.plotTreeWidget.dropSignal.connect(self.redrawAll)
+
+        self.setupMofifierCombobox()
+        self.plot_widget = PlotWidget(self.datacollector)
+        self.data_origin_fitpage_index = None
+
+    def create_plot(self, fitpage_index):
+        self.update_plot_tree(fitpage_index)
+        self.plot_widget.show()
+        self.plot_widget.activateWindow()
+
+    def update_datasets_from_collector(self):
+        # block signals to prevent currentItemChanged to be called. otherwise the program crashes, because it tries
+        # to access the current item.
+        self.dataTreeWidget.blockSignals(True)
+        self.dataTreeWidget.clear()
+        self.dataTreeWidget.blockSignals(False)
+
+        datasets = self.datacollector.get_datasets()
+        for i in range(len(datasets)):
+            fitpage_index = datasets[i].get_fitpage_index()
+            name = "Data from Fitpage " + str(fitpage_index)
+            data_id = datasets[i].get_data_id()
+            item = PlotPageItem(self.dataTreeWidget, [name], fitpage_index, data_id)
+            item.setData(0, 1, item)
+            subitem_data = DataItem(item, ["Data"], fitpage_index, data_id, 1)
+            subitem_data.setData(0, 1, subitem_data)
+            if datasets[i].has_y_fit():
+                subitem_fit = DataItem(item, ["Fit"], fitpage_index, data_id, 2)
+                subitem_fit.setData(0, 1, subitem_fit)
+
+        self.dataTreeWidget.expandAll()
+
+    def onShowDataViewer(self):
+        if self.isVisible():
+            self.hide()
+            self.main_window.cmdShowDataViewer.setText("Show Data Viewer")
+        else:
+            self.update_datasets_from_collector()
+            self.show()
+            self.main_window.cmdShowDataViewer.setText("Hide Data Viewer")
+
+    def update_dataset(self, main_window, fitpage_index, create_fit, checked_2d):
+        self.datacollector.update_dataset(main_window, fitpage_index, create_fit, checked_2d)
+
+    def update_plot_tree(self, fitpage_index):
+        # check if an item for the fitpage index already exists
+        # if one is found - remove from tree
+        for i in range(self.plotTreeWidget.topLevelItemCount()):
+            if isinstance(self.plotTreeWidget.topLevelItem(i), TabItem):
+                if fitpage_index == self.plotTreeWidget.topLevelItem(i).data(0, 1).get_fitpage_index():
```

Comment:
This could be all on one line. Python evaluates each part of an if statement individually and from left to right, so there will be no cost or increase in errors by combining these.

`if isinstance(self.plotTreeWidget.topLevelItem(i), TabItem) and fitpage_index == self.plotTreeWidget.topLevelItem(i).data(0, 1).get_fitpage_index():`